hedera-mirror-rest/__tests__/service/contractService.test.js

```diff
@@ -216,3 +216,220 @@ describe('ContractService.getContractResultsByIdAndFilters tests', () => {
     expect(response).toMatchObject(expectedContractResult);
   });
 });
+
+describe('ContractService.getContractResultsByIdAndTimestamp tests', () => {
+  test('ContractService.getContractResultsByIdAndTimestamp - No match', async () => {
+    await expect(ContractService.getContractResultsByIdAndTimestamp()).resolves.toEqual(null);
+  });
+
+  test('ContractService.getContractResultsByIdAndTimestamp - Row match', async () => {
+    await integrationDomainOps.loadContractResults([
+      {
+        contract_id: 2,
+        consensus_timestamp: 2,
+        function_parameters: '\\x0D',
+        amount: 10,
+        payer_account_id: '5',
+      },
+    ]);
+
+    await integrationDomainOps.loadRecordFiles([
+      {
+        index: 1,
+        consensus_start: 1,
+        consensus_end: 3,
+        hash: 'dee34',
+      },
+    ]);
+
+    await integrationDomainOps.loadTransactions([
+      {
+        consensus_timestamp: 2,
+        payerAccountId: '5',
+      },
+    ]);
+
+    const expectedDetailedContractResult = {
+      contractResult: {
+        amount: '10',
+        bloom: null,
+        callResult: null,
+        consensusTimestamp: '2',
+        contractId: '2',
+        createdContractIds: [],
+        errorMessage: '',
+        functionResult: null,
+        gasLimit: '1000',
+        gasUsed: '10',
+        payerAccountId: '5',
+      },
+      recordFile: {
+        consensusEnd: '3',
+        hash: 'dee34',
+        index: '1',
+      },
+      transaction: {
+        consensusTimestamp: '2',
+        payerAccountId: '5',
+      },
+    };
+
+    await expect(ContractService.getContractResultsByIdAndTimestamp(2, 2)).resolves.toMatchObject(
```

Comment:
```suggestion
    await expect(ContractService.getContractResultsByIdAndTimestamp(2, 2)).resolves.toEqual(
```

toMatchObject allows the `actual` to be a superset of the `expected`

Reply:
Yes, that's the intention. So you don't have to check every field but can verify on the uniquely identifiable and appropriate columns for the test.
Some fields are left out of input for brevity